plugins/solidigm/solidigm-log-page-dir.c

```diff
@@ -0,0 +1,292 @@
+		{ 0x0E, "Predictable Latency Event Aggregate"},
+		{ 0x0F, "Endurance Group Event Aggregate"},
+		{ 0x10, "Media Unit Status"},
+		{ 0x11, "Supported Capacity Configuration List"},
+		{ 0x12, "Feature Identifiers Supported and Effects"},
+		{ 0x13, "NVMe-MI Commands Supported and Effects"},
+		{ 0x14, "Command and Feature lockdown"},
+		{ 0x15, "Boot Partition"},
+		{ 0x16, "Rotational Media Information"},
+		{ 0x70, "Discovery"},
+		{ 0x80, "Reservation Notification"},
+		{ 0x81, "Sanitize Status"},
+	};
+	const int standard_log_size = sizeof(standard_log_descs) / sizeof(struct log_description);
+
+	static struct log_description ocp_log_descs[] = {
+		{ 0xC0, "OCP SMART / Health Information Extended" },
+		{ 0xC1, "OCP Error Recovery" },
+		{ 0xC2, "OCP Firmware Activation History" },
+		{ 0xC3, "OCP Latency Monitor" },
+		{ 0xC4, "OCP Device Capabilities" },
+		{ 0xC5, "OCP Unsupported Requirements" },
+	};
+	const int ocp_log_size = sizeof(ocp_log_descs) / sizeof(struct log_description);
+
+	static struct log_description solidigm_log_descs[] = {
+		{ 0xC1, "Read Commands Latency Statistics" },
+		{ 0xC2, "Write Commands Latency Statistics" },
+		{ 0xC4, "Endurance Manager Statistics" },
+		{ 0xC5, "Temperature Statistics" },
+		{ 0xCA, "SMART Attributes" },
+	};
+	const int solidigm_log_size = sizeof(solidigm_log_descs) / sizeof(struct log_description);
+
+	static struct log_description all_vu_log_descs[] = {
+		{ 0xC0, "OCP SMART / Health Information Extended" },
+		{ 0xC1, "OCP Error Recovery or Read Commands Latency Statistics" },
+		{ 0xC2, "OCP Firmware Activation History or Write Commands Latency Statistics" },
+		{ 0xC3, "OCP Latency Monitor" },
+		{ 0xC4, "OCP Device Capabilities or Endurance Manager Statistics" },
+		{ 0xC5, "OCP Unsupported Requirements or Temperature Statistics" },
+		{ 0xCA, "SMART Attributes" },
+	};
+	const int all_vu_log_size = sizeof(all_vu_log_descs) / sizeof(struct log_description);
+
+	// Standard logs are less than 0xC0
+	if (lid < 0xC0)
+		return lid_desc_from_struct(lid, standard_log_size, standard_log_descs);
+	else if (uuid == OCP_UUID)
+		return lid_desc_from_struct(lid, ocp_log_size, ocp_log_descs);
+	// Otherwise these are Solidigm logs.
+	else if (uuid == SOLIDIGM_UUID)
+		return lid_desc_from_struct(lid, solidigm_log_size, solidigm_log_descs);
+	else if (uuid == NO_UUID)
+		return lid_desc_from_struct(lid, all_vu_log_size, all_vu_log_descs);
+
+	return "Unknown";
+}
+
+static void solidigm_supported_log_pages_print(const struct supported_log_pages *supported,
+					       const enum solidigm_uuid uuid)
+{
+	printf("Log Page Directory Log:\n");
+	printf("  Supported:\n");
+
+	for (int lid = 0; lid < sizeof(supported->supported) / sizeof(__u32); lid++) {
+		if (supported->supported[lid] == 0)
+			continue;
+
+		printf("    Log Page:\n");
+		printf("      %-16s0x%02x\n", "LID:", le32_to_cpu(lid));
+		printf("      %-16s%s\n", "Description:", lid_to_desc(lid, uuid));
+	}
+
+	printf("\n");
```

Comment:
Maybe flow the outformat of /nvme supported-log-pages

Reply:
I'm not sure what you mean here. Do you mean follow instead of flow?

Reply:
Ooops,  I meant to say:
Maybe follow the output format of "nvme supported-log-pages"

Reply:
We are adding extra info (I think) but sure I will take a look

Reply:
Updated to a table output